testsuites/syncgateway/performance/run_sgload_perf_test.py

```diff
@@ -0,0 +1,176 @@
+        log_error(traceback.format_exc())
+        raise e
+
+
+def get_load_generators_hosts(cluster_config):
+    # Get load generator ips from ansible inventory
+    return get_hosts_by_type(cluster_config, "load_generators")
+
+
+def get_sync_gateways_hosts(cluster_config):
+    # Get sync gateway ips from ansible inventory
+    return get_hosts_by_type(cluster_config, "sync_gateways")
+
+
+def get_hosts_by_type(cluster_config, host_type="load_generators"):
+    lgs_hosts = hosts_for_tag(cluster_config, host_type)
+    lgs = [lg["ansible_host"] for lg in lgs_hosts]
+    return lgs
+
+
+def add_sync_gateway_url(cluster_config, sgload_arg_list):
+    """
+    Add ['--sg-url', 'http://..'] to the list of args that will be passed to sgload
+    """
+    sg_hosts = get_sync_gateways_hosts(cluster_config)
+    if len(sg_hosts) == 0:
+        raise Exception("Did not find any SG hosts")
+    sgload_arg_list.append("--sg-url")
+    sgload_arg_list.append("http://{}:4984/db/".format(sg_hosts[0]))  # TODO: don't hardcode port or DB name
+    return sgload_arg_list
+
+
+def get_lg2sg_map(lg_hosts, sg_hosts):
+    """
+    Assign load generators to sync gateways and return a map
+    with the assignments where key=lg, val=sg:
+
+    {'192.168.33.13': '192.168.33.11'}
+    """
+
+    lg2sgmap = {}
+    for lg_host in lg_hosts:
+        r = random.Random()
+        highest_sg_index = len(sg_hosts) - 1
+        sg_index = r.randint(0, highest_sg_index)
+        sg_host = sg_hosts[sg_index]
+        lg2sgmap[lg_host] = sg_host
+    return lg2sgmap
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--skip-build-sgload', action='store_true')
+    args = parser.parse_known_args()
+    known_args, sgload_arg_list_main = args  # unroll this tuple into named args
+
+    try:
+        main_cluster_config = os.environ["CLUSTER_CONFIG"]
+    except KeyError:
+        print ("Make sure CLUSTER_CONFIG is defined and pointing to the configuration you would like to provision")
+        sys.exit(1)
+
+    sgload_arg_list_main = add_sync_gateway_url(main_cluster_config, sgload_arg_list_main)
+
+    print("Running sgload perf test against cluster: {}".format(main_cluster_config))
+    main_ansible_runner = AnsibleRunner(main_cluster_config)
+
+    # Install + configure telegraf
+    status = main_ansible_runner.run_ansible_playbook("install-telegraf.yml")
+    if status != 0:
+        raise ProvisioningError("Failed to install telegraf")
+
+    # build_sgload (ansible)
+    if not known_args.skip_build_sgload:
+        build_sgload(main_ansible_runner)
+
+    # get load generator and sg hostnames
+    lg_hosts_main = get_load_generators_hosts(main_cluster_config)
+    sg_hosts_main = get_sync_gateways_hosts(main_cluster_config)
+
+    # Get a map from load generator hostnames to sync gateway hostnames
+    # eg, {'192.168.33.13': '192.168.33.11'} key=lg, val=sg
+    lg2sg = get_lg2sg_map(lg_hosts_main, sg_hosts_main)
```

Comment:
Fixed in https://github.com/couchbaselabs/mobile-testkit/pull/743/commits/b1570557525cb9c47ea61ce19477cf1760d4cb08